Tests/SecureXPCTests/XPCEncoder/Scalar encoding tests.swift

```diff
@@ -0,0 +1,240 @@
+		try assert(0.0 as Float,
+				   encodesEqualTo: xpc_double_create(0.0))
+		try assert(123.0 as Float,
+				   encodesEqualTo: xpc_double_create(123.0))
+		try assert(Double.greatestFiniteMagnitude,
+				   encodesEqualTo: xpc_double_create(Double.greatestFiniteMagnitude))
+		try assert(Double.infinity,
+				   encodesEqualTo: xpc_double_create(Double.infinity))
+		
+		// NaN
+		let result = xpc_double_get_value(try encode(Double.nan))
+		XCTAssert(result.isNaN)
+	}
+	
+	func testEncodes_Double_SignalingNaN_as_XPCDouble_SignalingNaN() throws {
+		let result = xpc_double_get_value(try encode(Double.signalingNaN))
+		XCTAssert(result.isNaN)
+		XCTAssert(result.isSignalingNaN)
+	}
+	
+	// MARK: Misc. types
+	
+	func testEncodes_Bool_as_XPCBool() throws {
+		try assert(false, encodesEqualTo: xpc_bool_create(false))
+		try assert(true, encodesEqualTo: xpc_bool_create(true))
+		try assert(false, encodesEqualTo: XPC_BOOL_FALSE)
+		try assert(true, encodesEqualTo: XPC_BOOL_TRUE)
+	}
+	
+	func testEncodes_Data_as_XPCData() throws { // 💾
+		throw XCTSkip("Support for Data is not implemented, the contents just end up as a String.")
+		
+		let emptyData = Data()
+		
+		try emptyData.withUnsafeBytes { buffer in
+			try assert(emptyData, encodesEqualTo: xpc_data_create(buffer.baseAddress!, buffer.count))
+		}
+		
+		let sampleData = "Hello, world!".data(using: .utf8)!
+		
+		try sampleData.withUnsafeBytes { buffer in
+			try assert(sampleData, encodesEqualTo: xpc_data_create(buffer.baseAddress!, buffer.count))
+		}
+	}
+	
+	func testEncodes_Date_as_XPCDate() throws { // 📆
+		throw XCTSkip("Support for Date is not implement, the content just end up as a Double.")
+		
+		let now = Date()
+		
+		// TODO: the conversion from Double to Int64 might cause some rounding that we'd need to account for here
+		try assert(Date.distantPast,   encodesEqualTo: xpc_date_create(Int64(Date.distantPast  .timeIntervalSince1970)))
+		try assert(now,                encodesEqualTo: xpc_date_create(Int64(               now.timeIntervalSince1970)))
+		try assert(Date.distantFuture, encodesEqualTo: xpc_date_create(Int64(Date.distantFuture.timeIntervalSince1970)))
+	}
+	
+	func testEncodes_String_as_XPCString() throws {
+		try assert("", encodesEqualTo: xpc_string_create(""))
+		try assert("Hello, world!", encodesEqualTo: xpc_string_create("Hello, world!"))
+	}
+	
+	func testEncodes_UUID_as_XPCUUID() throws {
+		throw XCTSkip("Support for UUID is not implemented, the contents just end up as a String.")
+		
+		let sampleUUID = UUID(uuidString: "34933b03-769c-4e50-a44a-5335f584aa9a")!
+		
+		// If you squint a little, you can see it's the same as the string above ;)
+		let expectedXPCUUID = xpc_uuid_create([
+			0x34,0x93,0x3b,0x03, // -
+			0x76,0x9c, // -
+			0x4e,0x50, // -
+			0xa4,0x4a, // -
+			0x53,0x35,0xf5,0x84,0xaa,0x9a,
+		])
+		
+		try assert(sampleUUID, encodesEqualTo: expectedXPCUUID)
+	}
+	
+	func testEncodes_NilValues_as_XPCNull() throws {
```

Comment:
Hmm, is there any value in doing this for all of these different types? It seems like what this is actually testing is Swift's implementation of `Optional`?

Reply:
In effect; kind of. But that's not something that I knew from black box testing this (I wrote as much of this as possible without biasing myself by looking at the implementation)